### PR TITLE
DOC/MAINT: linalg: fix some typos

### PR DESCRIPTION
**Pull request by musvaage**
@lucascolley

dumped directory strings exposed a few typos

### CODE REVIEW

File: scipy/linalg/_decomp.py [hunk 1]
```diff
@@ -793,13 +793,13 @@ def eig_banded(a_band, lower=False, eigvals_only=False, overwrite_a_band=False,
     del select_range
     if select == 0:
         if a1.dtype.char in 'GFD':
-            # FIXME: implement this somewhen, for now go with builtin values
+            # FIXME: implement this somehow, for now go with builtin values
```
Comment (lucascolley) [inline scipy/linalg/_decomp.py:796]: I would revert this - the intention was "at some other time"

**Review by lucascolley — COMMENTED** [review]
all good modulo one file

**Review by rkern — COMMENTED** [review]
The other changes LGTM.

**Comment by lucascolley** [comment]
snap :)

File: scipy/linalg/_decomp.py [hunk 2]
```diff
@@ -793,13 +793,13 @@ def eig_banded(a_band, lower=False, eigvals_only=False, overwrite_a_band=False,
     del select_range
     if select == 0:
         if a1.dtype.char in 'GFD':
-            # FIXME: implement this somewhen, for now go with builtin values
```
[the remark this answers is not in the document]
Comment (musvaage) [reply scipy/linalg/_decomp.py:796]: Do you mind a soft reset after that solitary fix?

**Review by lucascolley — APPROVED** [review]
thanks!

**Comment by musvaage** [comment]
In the scipy/optimize tree any sudirectories to skip or specific file types regarding typo checking?

**Comment by lucascolley** [comment]
consult gh-21232 for vendored code that shouldn't be touched.

**Pull request merged**